aggregator/aggregator.go

```diff
@@ -111,45 +111,41 @@ var _ core.Metricable = (*Aggregator)(nil)
 // TODO: Remove this context once OperatorPubkeysServiceInMemory's API is
 // changed and we can gracefully exit otherwise
 func NewAggregator(ctx context.Context, config *config.Config, logger logging.Logger) (*Aggregator, error) {
-	chainioConfig := eigenclients.BuildAllConfig{
-		EthHttpUrl:                 config.EthHttpRpcUrl,
-		EthWsUrl:                   config.EthWsRpcUrl,
-		RegistryCoordinatorAddr:    config.SFFLRegistryCoordinatorAddr.String(),
-		OperatorStateRetrieverAddr: config.OperatorStateRetrieverAddr.String(),
-		AvsName:                    avsName,
-		PromMetricsIpPortAddress:   config.MetricsIpPortAddress,
-	}
-	clients, err := eigenclients.BuildAll(chainioConfig, config.EcdsaPrivateKey, logger)
-	if err != nil {
-		logger.Error("Cannot create sdk clients", "err", err)
-		return nil, err
+	withOptionalMetrics := func(url string) safeclient.SafeEthClientOption {
+		if !config.EnableMetrics {
+			return func(*safeclient.SafeEthClient) {}
+		}
+
+		registry := prometheus.NewRegistry()
+		rpcCallsCollector := rpccalls.NewCollector(AggregatorNamespace+url, registry)
```

Comment:
On each usage of `withOptionalMetrics` we're creating a new `registry` to maintain the previous behavior